crates/dbsp/src/storage/dirlock/mod.rs

```diff
@@ -0,0 +1,168 @@
+//! A simple PID-based locking mechanism.
```

Comment:
common, isn't there a Rust library for this?
I recall I saw Ben write something at some point

Reply:
there are some but none convinced me.. I also wanted to avoid adding yet another dependency for a few 100 lines of code to do some good for our project